### ISSUE TITLE
`@override` does not work correctly for overloaded method without implementation

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#9746]
If an overloaded method has no implementation (e.g. it's in a stub file, protocol or ABC) and the first overloaded method is decorated with `@override`, this override isn't enforced. 

```python
class Base(Protocol):
    pass


class Subclass(Base, Protocol):
    @overload
    @override
    # This should generate an error because method isn't present in the base.
    def method(self, x: int) -> int: ...
    @overload
    def method2(self, x: str) -> str: ...
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.393.
